portals/admin/source/src/app/components/AdvancedSettings/TenantConfSave.jsx

```diff
@@ -0,0 +1,240 @@
+ */
+
+import React, {
+    useReducer, useState, Suspense, lazy, useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useIntl, FormattedMessage } from 'react-intl';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    Box, Grid, Button,
+} from '@material-ui/core';
+import { green } from '@material-ui/core/colors';
+import API from 'AppData/api';
+import Alert from 'AppComponents/Shared/Alert';
+import { Progress } from 'AppComponents/Shared';
+import ContentBase from 'AppComponents/AdminPages/Addons/ContentBase';
+
+const MonacoEditor = lazy(() => import('react-monaco-editor' /* webpackChunkName: "TeantConfAddMonacoEditor" */));
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        marginBottom: theme.spacing(15),
+    },
+    error: {
+        color: theme.palette.error.dark,
+    },
+    dialog: {
+        minWidth: theme.spacing(150),
+
+    },
+    tenantConfHeading: {
+        marginBottom: theme.spacing(1),
+    },
+    showSampleButton: {
+        marginTop: theme.spacing(2),
+    },
+    helperText: {
+        color: green[600],
+        fontSize: theme.spacing(1.6),
+        marginLeft: theme.spacing(1),
+    },
+    infoBox: {
+        marginTop: theme.spacing(1),
+        marginBottom: theme.spacing(2),
+    },
+    buttonBox: {
+        marginTop: theme.spacing(2),
+        marginBottom: theme.spacing(2),
+        marginLeft: theme.spacing(1),
+    },
+    saveButton: {
+        marginRight: theme.spacing(2),
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state The second number.
+ * @returns {Promise}
+ */
+function reducer(state, { field, value }) {
+    switch (field) {
+        case 'tenantConf':
+            return { ...state, [field]: value };
+        case 'tenantConfSchema':
+            return { ...state, [field]: value };
+        case 'editDetails':
+            return value;
+        default:
+            return state;
+    }
+}
+
+/**
+ * Render a list
+ * @returns {JSX} Header AppBar components.
+ */
+function TenantConfSave(props) {
+    const classes = useStyles();
+    const {
+        updateList, history,
+    } = props;
+    const intl = useIntl();
+    const [initialState, setInitialState] = useState({
+        tenantConf: '',
+        tenantConfSchema: '',
+    });
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const {
+        tenantConf, tenantConfSchema,
+    } = state;
+    const restApi = new API();
+
+    useEffect(() => {
+        let tenantConfVal;
+        let tenantConfSchemaVal;
+        restApi.tenantConfSchemaGet().then((result) => {
+            tenantConfSchemaVal = result.body;
+            dispatch({ field: 'tenantConfSchema', value: tenantConfSchemaVal });
+        });
+        restApi.tenantConfGet().then((result) => {
+            tenantConfVal = JSON.stringify(result.body, null, '\t');
+            dispatch({ field: 'tenantConf', value: tenantConfVal });
+        });
+        setInitialState({
```

Comment:
This has no effect on the reducer's initial state

Reply:
will change accordingly

Reply:
resolved